libraries/psibase/common/include/psibase/jwt.hpp

```diff
@@ -9,12 +10,21 @@ namespace psibase
    using JWTKey = std::span<const char>;
 
    std::string encodeJWT(const JWTKey&, std::string_view json);
-   std::string decodeJWT(const JWTKey&, std::string_view token);
+   void        validateJWT(const JWTKey&, std::string_view token);
+   std::string decodeJWTPayloadUnsafe(std::string_view token);
 
    template <typename T>
-   T decodeJWT(const JWTKey& key, std::string_view token)
+   std::optional<T> decodeJWT(const JWTKey&                 key,
+                              std::string_view              token,
+                              std::function<bool(const T&)> validatePayload)
    {
-      return psio::convert_from_json<T>(decodeJWT(key, token));
+      auto payload = psio::convert_from_json<T>(decodeJWTPayloadUnsafe(token));
+      if (validatePayload(payload))
```

Comment:
This is kind of backwards. I think it would make more sense to verify the MAC without aborting. If we ever decide to use encrypted tokens, decoding a token without the correct key isn't even possible.

Reply:
> If we ever decide to use encrypted tokens, decoding a token without the correct key isn't even possible.

Good point.

So wait, your proposal here is that if validation fails, then don't abort, and don't return an http error, simply treat user = None. Right?

But if the validation succeeds, and then I decode the token and find that it's expired, then should that be an http error?

Reply:
> and don't return an http error

I could go either way on that.